### PR TITLE
Add SentryLog when mailboxId uses the default value for configuration

### PR DESCRIPTION
**Pull request by TommyDL-Infomaniak**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 1]
```diff
@@ -229,12 +231,15 @@ object RealmDatabase {
...
             .build()
 
-        fun mailboxContent(mailboxId: Int, userId: Int) = RealmConfiguration
-            .Builder(mailboxContentSet)
-            .name(mailboxContentDbName(userId, mailboxId))
-            .schemaVersion(MAILBOX_CONTENT_SCHEMA_VERSION)
-            .migration(MAILBOX_CONTENT_MIGRATION)
-            .build()
+        fun mailboxContent(mailboxId: Int, userId: Int): RealmConfiguration {
+            if (mailboxId == DEFAULT_ID) SentryLog.e(TAG, "RealmConfiguration problem with mailbox content, default ID is used.")
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:235]: This SentryLog won't show by itself in Sentry.
You should replace it by a `Sentry.captureMessage`

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 2]
```diff
@@ -229,12 +232,18 @@ object RealmDatabase {
...
             .build()
 
-        fun mailboxContent(mailboxId: Int, userId: Int) = RealmConfiguration
-            .Builder(mailboxContentSet)
-            .name(mailboxContentDbName(userId, mailboxId))
-            .schemaVersion(MAILBOX_CONTENT_SCHEMA_VERSION)
-            .migration(MAILBOX_CONTENT_MIGRATION)
-            .build()
+        fun mailboxContent(mailboxId: Int, userId: Int): RealmConfiguration {
+            if (mailboxId == DEFAULT_ID) Sentry.captureMessage(
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:236]:
```suggestion

            if (mailboxId == DEFAULT_ID) Sentry.captureMessage(
```

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 3]
```diff
@@ -229,12 +232,18 @@ object RealmDatabase {
...
+                "RealmConfiguration problem with mailbox content, default ID is used.",
+                SentryLevel.ERROR
+            )
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:239]: Add curly braces to this `if`.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 4]
```diff
@@ -139,7 +142,7 @@ object RealmDatabase {
 
     //region Delete Realm
     fun deleteMailboxContent(mailboxId: Int, userId: Int = AccountUtils.currentUserId) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt:144]: Invert here too.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt [hunk 5]
```diff
@@ -229,12 +232,18 @@ object RealmDatabase {
...
-        fun mailboxContent(mailboxId: Int, userId: Int) = RealmConfiguration
-            .Builder(mailboxContentSet)
-            .name(mailboxContentDbName(userId, mailboxId))
-            .schemaVersion(MAILBOX_CONTENT_SCHEMA_VERSION)
-            .migration(MAILBOX_CONTENT_MIGRATION)
-            .build()
+        fun mailboxContent(mailboxId: Int, userId: Int): RealmConfiguration {
+            if (mailboxId == DEFAULT_ID) Sentry.captureMessage(
+                "RealmConfiguration problem with mailbox content, default ID is used.",
+                SentryLevel.ERROR
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/cache/RealmDatabase.kt]: Code is outdated, doesn't make sense anymore.

**Review by KevinBoulongne — APPROVED** [review]

**Review by FabianDevel — APPROVED** [review]

**Pull request merged**